### PR TITLE
Blog-app: Validation

### PR DESCRIPTION
**Pull request by mahdinoori2000**
### What is added

- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by zieeco — COMMENTED** [review]
# STATUS: INVALID!!!  :x: :x: :x: 

Hi @mahdinoori2000,




While you made a great effort in this project, unfortunately, I cannot proceed to review your code, please make sure you implement the following changes before you resubmit for review. 

## Required Changes ♻️
_Check the comments below._

- [ ] As it is a requirement for this project, please kindly fix the linter error.


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Invalid Code Review Request **Does not count** into the code reviews limit._
# **INVALID PROJECT**

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    expect(user.errors[:posts_counter]).to include('is not a number')
+  end
+
+  it 'should return the post_counter number correctly' do
+    user = User.create(posts_counter: 5)
+    user.valid?
+    expect(user.posts_counter).to eq(5)
+  end
+
+  it 'returns three most recent posts' do
```
Comment (FrederickMih) [inline spec/user_spec.rb]: - [ ] As you can see on the screenshot below, some of your test units are failing. Please make sure to fix :wrench:  them :+1: 
Screenshot from 2023-10-07 22-53-36

File: spec/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    user = User.create(name: 'Test User')
+    post1 = user.posts.create(title: '1st post')
+    post2 = user.posts.create(title: '2nd post')
+    post3 = user.posts.create(title: '3rd post')
+
+    recent_posts = user.three_recent_posts
+
+    expect(recent_posts).to eq([post3, post2, post1])
+  end
```
Comment (FrederickMih) [inline spec/user_spec.rb]: - [ ] In the course of development, to actually show that this method returns 3 most recent posts of the user, we were asked to create at least 4 posts. Calling this method on the posts should return only the recent 3 posts created by the user. Hence I think you should do same here. Kindly create 4 or more posts and show that only the 3 most recent posts are expected :+1:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @mahdinoori2000, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Awesome job for specifying test type: This way, RSpec Rails understand what type of spec we are running 🚀 👍


## Required Changes ♻️

- [ ] The MIT license link is not working. Please kindly fix the link in order to render README.md file professional. Broken links hurt user's experience.
Screenshot from 2023-10-07 22-32-56


_Check the comments under the review._

## Optional suggestions

-  Organizing your files in a project makes it easier for you and others to find and use them. Since you'll required to also write specs for controllers and views, I think you should move and place your spec files into the model folder :+1: 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — COMMENTED** [review]
# Hi @mahdinoori2000 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- Hey Mahdi, I wanted to remind you about Microverse's code review policy. Since your previous PR for the current week's assignment was not approved and you are repeating the week, you'll have to create a new PR for your updated work. Once you have done that, please submit the link to the new PR as per Microverse's code review guidelines. This will ensure that your work receives the necessary review and feedback. Thank you for your attention to this matter.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Pull request closed without merging**